api/dataprocessor.go

```diff
@@ -166,22 +147,22 @@ func aggEvery(numPoints, maxPoints uint32) uint32 {
 }
 
 // error is the error of the first failing target request
-func getTargets(store mdata.Store, reqs []Req) ([]Series, error) {
-	seriesChan := make(chan Series, len(reqs))
+func (s *Server) getTargets(reqs []models.Req) ([]models.Series, error) {
+	seriesChan := make(chan models.Series, len(reqs))
 	errorsChan := make(chan error, len(reqs))
 	// TODO: abort pending requests on error, maybe use context, maybe timeouts too
 	wg := sync.WaitGroup{}
 	wg.Add(len(reqs))
 	for _, req := range reqs {
-		go func(wg *sync.WaitGroup, req Req) {
+		go func(wg *sync.WaitGroup, req models.Req) {
 			pre := time.Now()
-			points, interval, err := getTarget(store, req)
+			points, interval, err := s.getTarget(req)
 			if err != nil {
 				errorsChan <- err
```

Comment:
panic should only be called when an error is encountered that you can not recover from.  if the error is expected and can be handled then just dont panic.  Having to write too much code is not a justification for using panic instead of handling errors properly.

Ill put doRecover back, but this will all need to be cleaned up. 

And please stop calling panic.  If an error is fatal and you want to terminate the process, call log.Fatal() instead.